### PR TITLE
Add e2e tests

### PR DESCRIPTION
**Pull request by delfrrr**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -26,6 +26,21 @@ RUN go build ./src/server
...
+ENV DEKART_PORT=3000
+ENV DEKART_STATIC_FILES=./build
+COPY --from=nodebuilder /source/build build
+COPY --from=gobuilder /source/server .
+ADD migrations migrations
+ADD cypress cypress
+ADD cypress.config.js .
+ADD package.json .
+ENTRYPOINT ["/bin/sh", "-c" , "/dekart/server & cypress run"]
+
```
Comment (Tsovak) [inline Dockerfile:43]: create a separate dockerfile for tests and cypress

File: cypress/e2e/spec.cy.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+/* global cy */
+
+describe('visit main page', () => {
+  it('make simple bigquery query', () => {
+    cy.visit('/')
+    cy.get('button:contains("Create Report")').click()
```
Comment (Tsovak) [inline cypress/e2e/spec.cy.js:6]: can we use xpath or id to find a button? the text can be changed and tests will fail

**Review by Tsovak — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -16,28 +16,45 @@ jobs:
...
+      #   with:
+      #     context: .
+      #     target: gobuilder
+      #     cache-from: type=gha
+      #     cache-to: type=gha,mode=max
+      # - name: Client Build
+      #   uses: docker/build-push-action@v2
+      #   with:
+      #     context: .
+      #     target: nodebuilder
```
[the remark this answers is not in the document]
Comment (delfrrr) [reply .github/workflows/test.yaml]: Was just testing pipelines; disable to make it faster; now setup cache for it

File: Dockerfile (hunk 1 above)
Comment (delfrrr) [reply Dockerfile:43]: Why? Then I cannot reuse cache and builds; I'm following concept of using docker stages to build and run tests

File: cypress/e2e/spec.cy.js (hunk 2 above)
Comment (delfrrr) [reply cypress/e2e/spec.cy.js:6]: I see pros and cons: If the button has no text, for example, then the user will not recognize but test still pass

**Pull request merged**